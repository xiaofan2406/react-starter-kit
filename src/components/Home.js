import React from 'react';
import { css } from '@emotion/core';
import { spacing } from 'styles';
import { Anchor } from 'widgets';

const Home = () => (
  <div>
    <div
      css={css`
        font-size: 48px;
        text-align: center;
        margin-bottom: ${spacing.breath}px;
      `}
    >
      Hello world!
    </div>
    <table
      css={css`
        margin-top: ${spacing.breath}px;
        & td {
          padding: ${spacing.internal}px;
        }
      `}
    >
      <tbody>
        <tr>
          <td>React</td>
          <td>
            <Anchor href="https://reactjs.org/docs/thinking-in-react.html">
              Docs
            </Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/facebook/react/">Github</Anchor>
          </td>
        </tr>
        <tr>
          <td>Reach Router</td>
          <td>
            <Anchor href="https://reach.tech/router">Docs</Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/reach/router">Github</Anchor>
          </td>
        </tr>
        <tr>
          <td>Emotion</td>
          <td>
            <Anchor href="https://emotion.sh/docs/composition">Docs</Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/emotion-js/emotion">Github</Anchor>
          </td>
        </tr>
        <tr>
          <td>Create React App</td>
          <td>
            <Anchor href="https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#table-of-contents">
              Docs
            </Anchor>
          </td>
          <td>
            <Anchor href="https://github.com/facebook/create-react-app/tree/next/packages/react-scripts">
              Github
            </Anchor>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
);

export default Home;
