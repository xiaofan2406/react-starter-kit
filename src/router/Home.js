import React from 'react';
import { css } from 'emotion';

const cssClass = css`
  font-size: 48px;
`;

const Home = () => <div className={cssClass}>Hello world!</div>;

export default Home;
