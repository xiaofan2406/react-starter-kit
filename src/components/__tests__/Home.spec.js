/* @flow */
import React from 'react';
import { render } from 'react-testing-library';
import Home from '../Home';

test('Home component matches snapshot', () => {
  const { container } = render(<Home />);

  expect(container.firstChild).toMatchSnapshot();
});
