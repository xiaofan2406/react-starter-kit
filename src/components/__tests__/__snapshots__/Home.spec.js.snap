// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
.emotion-0 > .title {
  font-size: 48px;
  text-align: center;
  margin-bottom: 24px;
}

.emotion-0 > table td {
  padding: 12px;
}

<div
  class="emotion-0"
>
  <div
    class="title"
  >
    Hello world!
  </div>
  <table>
    <tbody>
      <tr>
        <td>
          React
        </td>
        <td>
          <a
            href="https://reactjs.org/docs/thinking-in-react.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </td>
        <td>
          <a
            href="https://github.com/facebook/react/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github
          </a>
        </td>
      </tr>
      <tr>
        <td>
          React Router v4
        </td>
        <td>
          <a
            href="https://reacttraining.com/react-router/web/example/basic"
            rel="noopener noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </td>
        <td>
          <a
            href="https://github.com/ReactTraining/react-router"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github
          </a>
        </td>
      </tr>
      <tr>
        <td>
          Redux
        </td>
        <td>
          <a
            href="https://redux.js.org/introduction/three-principles"
            rel="noopener noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </td>
        <td>
          <a
            href="https://github.com/reactjs/redux"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github
          </a>
        </td>
      </tr>
      <tr>
        <td>
          Mobx
        </td>
        <td>
          <a
            href="https://mobx.js.org/best/react-performance.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </td>
        <td>
          <a
            href="https://github.com/mobxjs/mobx"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github
          </a>
        </td>
      </tr>
      <tr>
        <td>
          Emotion
        </td>
        <td>
          <a
            href="https://emotion.sh/docs/composition"
            rel="noopener noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </td>
        <td>
          <a
            href="https://github.com/emotion-js/emotion"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github
          </a>
        </td>
      </tr>
      <tr>
        <td>
          Create React App
        </td>
        <td>
          <a
            href="https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#table-of-contents"
            rel="noopener noreferrer"
            target="_blank"
          >
            Docs
          </a>
        </td>
        <td>
          <a
            href="https://github.com/facebook/create-react-app/tree/next/packages/react-scripts"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
